### PR TITLE
[api] add tracing events for slow api 

### PR DESCRIPTION
**Pull request by millken**

### CODE REVIEW

File: actpool/actpool.go [hunk 1]
```diff
@@ -423,9 +430,11 @@ func (ap *actPool) enqueueAction(addr address.Address, act action.SealedEnvelope
...
 	}
 
+	span.AddEvent("act.IntrinsicGas")
 	intrinsicGas, _ := act.IntrinsicGas()
 	ap.gasInPool += intrinsicGas
 	// If the pending nonce equals this nonce, update queue
+	span.AddEvent("queue.PendingNonce")
 	nonce := queue.PendingNonce()
 	if actNonce == nonce {
 		ap.updateAccount(sender)
```
Comment (Liuhaai) [inline actpool/actpool.go:440]: Could you add one more trace here?

**Review by Liuhaai — APPROVED** [review]

File: api/coreservice.go [hunk 2]
```diff
@@ -125,22 +125,27 @@ func newCoreService(
 }
 
 // Account returns the metadata of an account
-func (core *coreService) Account(addr address.Address) (*iotextypes.AccountMeta, *iotextypes.BlockIdentifier, error) {
+func (core *coreService) Account(ctx context.Context, addr address.Address) (*iotextypes.AccountMeta, *iotextypes.BlockIdentifier, error) {
```
Comment (CoderZhi) [inline api/coreservice.go:128]: revert

File: api/web3server_utils.go [hunk 3]
```diff
@@ -314,7 +314,7 @@ func (svr *Web3Server) isContractAddr(addr string) (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	accountMeta, _, err := svr.coreService.Account(ioAddr)
+	accountMeta, _, err := svr.coreService.Account(context.Background(), ioAddr)
```
Comment (CoderZhi) [inline api/web3server_utils.go:317]: revert

File: api/web3server.go [hunk 4]
```diff
@@ -371,7 +371,7 @@ func (svr *Web3Server) getBalance(in interface{}) (interface{}, error) {
 	if err != nil {
 		return nil, err
 	}
-	accountMeta, _, err := svr.coreService.Account(ioAddr)
+	accountMeta, _, err := svr.coreService.Account(context.Background(), ioAddr)
```
Comment (CoderZhi) [inline api/web3server.go:374]: revert

File: api/grpcserver.go [hunk 5]
```diff
@@ -101,14 +101,18 @@ func (svr *GRPCServer) SuggestGasPrice(ctx context.Context, in *iotexapi.Suggest
...
 // GetAccount returns the metadata of an account
 func (svr *GRPCServer) GetAccount(ctx context.Context, in *iotexapi.GetAccountRequest) (*iotexapi.GetAccountResponse, error) {
+	span := tracer.SpanFromContext(ctx)
+	defer span.End()
 	addr, err := address.FromString(in.Address)
 	if err != nil {
 		return nil, err
 	}
-	accountMeta, blockIdentifier, err := svr.coreService.Account(addr)
+	accountMeta, blockIdentifier, err := svr.coreService.Account(ctx, addr)
```
Comment (CoderZhi) [inline api/grpcserver.go:110]: revert

**Review by CoderZhi — CHANGES_REQUESTED** [review]

File: api/coreservice.go [hunk 6]
```diff
@@ -125,22 +125,27 @@ func newCoreService(
...
+	ctx, span := tracer.NewSpan(ctx, "coreService.Account")
```
[the remark this answers is not in the document]
Comment (millken) [reply api/coreservice.go:129]: If so, will not know who called it

**Review by CoderZhi — APPROVED** [review]

**Review by Scheusal13 — COMMENTED** [review]
io168vpa7fn9xk3jerkjf4mjgdmcrg3hv8af9qncm

**Pull request merged**